.github/workflows/check-kernel-config.yml

```diff
@@ -0,0 +1,74 @@
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the branch of our repo that triggered this action
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v2
+        
+      - name: Get base commit for PRs
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          git fetch origin ${{ github.base_ref }}
+          echo "base_sha=$(git rev-parse origin/${{ github.base_ref }})" >> $GITHUB_ENV
+          echo "Merging ${{ github.sha }} into ${{ github.base_ref }}"
+
+      - name: Get base commit for Pushes
+        if: ${{ github.event_name == 'push' }}
+        run: |
+          git fetch origin ${{ github.event.before }}
+          echo "base_sha=${{ github.event.before }}" >> $GITHUB_ENV
+          echo "Merging ${{ github.sha }} into ${{ github.event.before }}"
+
+
+      # For consistency, we use the same major/minor version of Python that CBL-Mariner ships
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Get Python dependencies
+        run: python3 -m pip install -r toolkit/scripts/requirements.txt
+
+      # Check kernel configs
+      - name: Get the changed files and configs
+        run: |
+          echo "Files changed: '$(git diff-tree --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }})'"
+          changed_configs=$(git diff-tree --diff-filter=d  --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS/kernel/config.*$" || test $? = 1; })
+          echo "Files to validate: '${changed_configs}'"
+          echo "updated-configs=$(echo ${changed_configs})" >> $GITHUB_ENV
+
+          # Check for new configs in the PR
+          # and check if they are in required configs
+          if [[ -n $changed_configs ]]; then
+            config_changes=$(git diff-tree -p -r ${{ env.base_sha }} ${{ github.sha }} -- SPECS/kernel/config)
+            echo "AMD config changes:"
+            python3 .github/workflows/check_new_kernel_configs.py --required_configs=toolkit/scripts/mariner-required-configs.json --config_str="${changed_configs}"
+            
+            echo "ARM config changes:"
+            config_aarch64_changes=$(git diff-tree -p -r ${{ env.base_sha }} ${{ github.sha }} -- SPECS/kernel/config_aarch64)
+            python3 .github/workflows/check_new_kernel_configs.py --required_configs=toolkit/scripts/mariner-required-configs.json --config_str="${config_aarch64_changes}"
+          fi
+
+      - name: Run kernel config checking script AMD64
+        run: python3 toolkit/scripts/check_required_kernel_configs.py --required_configs toolkit/scripts/mariner-required-configs.json --config_file SPECS/kernel/config
+      
+      - name: Run kernel config checking script ARM64
+        run: python3 toolkit/scripts/check_required_kernel_configs.py --required_configs toolkit/scripts/mariner-required-configs.json --config_file SPECS/kernel/config_aarch64
+
```

Comment:
Suggestion: it looks like we're first checking if any config files were touched in this PR and then go one-by-one inside the `if` statement anyway, even if only one of the configs was modified. While I believe that it's unlikely for only one of the configs to change at a time, I'm thinking we could make this code more flexible by moving away from the repeated steps in the `if` statement and just storing a list of configs we want to verify. Similarly, I think we could also completely drop the `changed_configs` variable:

```suggestion
    branches: [main, dev, 1.0*, 2.0*]

env:
  CHECKED_CONFIGS: ("SPECS/kernel/config SPECS/kernel/config_aarch64")

jobs:
  check:
    name: Kernel configs check
    runs-on: ubuntu-latest

    steps:
      # Checkout the branch of our repo that triggered this action
      - name: Workflow trigger checkout
        uses: actions/checkout@v2
        
      - name: Get base commit for PRs
        if: ${{ github.event_name == 'pull_request' }}
        run: |
          git fetch origin ${{ github.base_ref }}
          echo "base_sha=$(git rev-parse origin/${{ github.base_ref }})" >> $GITHUB_ENV
          echo "Merging ${{ github.sha }} into ${{ github.base_ref }}"

      - name: Get base commit for Pushes
        if: ${{ github.event_name == 'push' }}
        run: |
          git fetch origin ${{ github.event.before }}
          echo "base_sha=${{ github.event.before }}" >> $GITHUB_ENV
          echo "Merging ${{ github.sha }} into ${{ github.event.before }}"


      # For consistency, we use the same major/minor version of Python that CBL-Mariner ships
      - name: Setup Python 3.7
        uses: actions/setup-python@v2
        with:
          python-version: 3.7

      - name: Get Python dependencies
        run: python3 -m pip install -r toolkit/scripts/requirements.txt

      - name: Check new kernel configs entries
        run: |
          for checked_config in $CHECKED_CONFIGS; do
            echo "Checking for unexpected new entries inside ($checked_config)."

            config_changes=$(git diff-tree -p -r ${{ env.base_sha }} ${{ github.sha }} -- "$checked_config")
            python3 .github/workflows/check_new_kernel_configs.py --required_configs=toolkit/scripts/mariner-required-configs.json --config_str="${config_changes}"            
          done

      - name: Check required kernel config entries
        run: |
          for checked_config in $CHECKED_CONFIGS; do
            echo "Checking for required entries inside ($checked_config)."

            python3 toolkit/scripts/check_required_kernel_configs.py --required_configs toolkit/scripts/mariner-required-configs.json --config_file "$checked_config"
          done

```

Note 1: we could even use a matrix to run these in parallel for each config, but I think that's a bit of an overkill.
Note 2: I'm not entirely sure how failure is detected in these pipelines. It's possible we might need to use `set -e` or just exit manually whenever we detect a failure for steps that aren't just a single command.

Reply:
failures trigger a workflow failure :) I have seen enough now to know this :P